core/src/com/put/urbantraffic/CityGraph.java

```diff
@@ -0,0 +1,168 @@
+package com.put.urbantraffic;
+
+import com.put.urbantraffic.util.Pair;
+import lombok.Data;
+import lombok.val;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class CityGraph {
+    PathWithTime[][] generate(City city) {
+        val crossings = city.getCrossings();
+        val roads = city.getRoads();
+        val crossingsMap = crossings.stream()
+                .collect(Collectors.toMap(Crossing::getId, item -> item));
+        final Map<Pair<Integer, Integer>, Road> roadMap = roads.stream()
+                .collect(Collectors.toMap(road -> {
+                    val startId = road.getLaneList().get(0).getStartCrossing().getId();
+                    val endId = road.getLaneList().get(0).getEndCrossing().getId();
+                    return new Pair<>(startId, endId);
+                }, item -> item));
+
+        int crossingsAmount = crossings.size();
+
+        val neighbours = new ArrayList<List<CrossingWithTime>>(crossingsAmount);
+        for (int i = 0; i < crossingsAmount; i++) {
+            neighbours.add(new ArrayList<>());
+        }
+
+        // all neighbours
+        for (val road : roads) {
+            for (val lane : road.getLaneList()) {
+                val startCrossingId = lane.getStartCrossing().getId();
+                val endCrossing = lane.getEndCrossing();
+                float time = getTime(road);
+                neighbours.get(startCrossingId).add(new CrossingWithTime(endCrossing.getId(), time));
+            }
+        }
+
+        // run Dijkstra for each crossing
+        PathWithTime[][] result = new PathWithTime[crossingsAmount][];
+        for (int i = 0; i < crossingsAmount; i++) {
+            result[i] = convertIntoCityRepresentation(dijkstra(crossingsAmount, neighbours, i), crossingsMap, roadMap);
```

Comment:
I've split it into 2 functions as it makes it easier to remove unused representations

Reply:
w polsce jestes debilu

Reply:
nie nie mam dwubiegunowosci